File: CS/Models/XPAppointment.cs
using System;
using DevExpress.Xpo;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Xml;

public class XPAppointment : XPObject {
    public XPAppointment(Session session) : base(session) { }

    public bool AllDay;              // Appointment.AllDay

    [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
    public string Description;       // Appointment.Description

    public DateTime Finish;          // Appointment.End
    public int Label;                // Appointment.Label
    public string Location;          // Appointment.Location

    [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
    public string Recurrence;        // Appointment.RecurrenceInfo

    [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
    public string Reminder;          // Appointment.ReminderInfo

    public DateTime Created;         // Appointment.Start
    public int Status;               // Appointment.Status
    [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
    public string Subject;           // Appointment.Subject
    public int AppointmentType;      // Appointment.Type

    [Association()]
    public XPCollection<XPResource> Resources {
        get {
            return GetCollection<XPResource>("Resources");
        }
    }

    [NonPersistent()]
    public string ResourceIds {
        get {
            return GenerateResourceIdsString();
        }
        set {
            ResourceIdCollection resourceIds = GenerateResourceIdsString(value);
            Resources.SuspendChangedEvents();
            try {
                ClearResources();
                int count = resourceIds.Count;
                for (int i = 0; i < count; i++) {
                    XPResource resource = this.Session.GetObjectByKey<XPResource>(resourceIds[i]);
                    if (resource != null)
                        Resources.Add(resource);
                }
            }
            finally {
                Resources.ResumeChangedEvents();
            }
        }
    }

    void ClearResources() {
        int count = Resources.Count;
        while (count > 0) {
            Resources.Remove(Resources[0]);
            count--;
        }
    }
    ResourceIdCollection GenerateResourceIdsString(string xml) {
        ResourceIdCollection result = new ResourceIdCollection();
        if (String.IsNullOrEmpty(xml))
            return result;

        return AppointmentResourceIdCollectionXmlPersistenceHelper.ObjectFromXml(result, xml);
    }

    string GenerateResourceIdsString() {
        ResourceIdCollection resourceIds = new ResourceIdCollection();
        int count = Resources.Count;
        for (int i = 0; i < count; i++)
            resourceIds.Add(Resources[i].Oid);

        AppointmentResourceIdCollectionXmlPersistenceHelper helper = new AppointmentResourceIdCollectionXmlPersistenceHelper(resourceIds);
        return helper.ToXml();
    }
}

File: VB/Models/XPAppointment.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Xml

Public Class XPAppointment
	Inherits XPObject
	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Public AllDay As Boolean ' Appointment.AllDay

	<Size(SizeAttribute.Unlimited)> _
	Public Description As String ' Appointment.Description

	Public Finish As DateTime ' Appointment.End
	Public Label As Integer ' Appointment.Label
	Public Location As String ' Appointment.Location

	<Size(SizeAttribute.Unlimited)> _
	Public Recurrence As String ' Appointment.RecurrenceInfo

	<Size(SizeAttribute.Unlimited)> _
	Public Reminder As String ' Appointment.ReminderInfo

	Public Created As DateTime ' Appointment.Start
	Public Status As Integer ' Appointment.Status
	<Size(SizeAttribute.Unlimited)> _
	Public Subject As String ' Appointment.Subject
	Public AppointmentType As Integer ' Appointment.Type

	<Association()> _
	Public ReadOnly Property Resources() As XPCollection(Of XPResource)
		Get
			Return GetCollection(Of XPResource)("Resources")
		End Get
	End Property

	<NonPersistent()> _
	Public Property ResourceIds() As String
		Get
			Return GenerateResourceIdsString()
		End Get
		Set(ByVal value As String)
'INSTANT VB NOTE: The local variable resourceIds was renamed since Visual Basic will not allow local variables with the same name as their enclosing function or property:
			Dim resourceIds_Renamed As ResourceIdCollection = GenerateResourceIdsString(value)
			Resources.SuspendChangedEvents()
			Try
				ClearResources()
				Dim count As Integer = resourceIds_Renamed.Count
				For i As Integer = 0 To count - 1
					Dim resource As XPResource = Me.Session.GetObjectByKey(Of XPResource)(resourceIds_Renamed(i))
					If resource IsNot Nothing Then
						Resources.Add(resource)
					End If
				Next i
			Finally
				Resources.ResumeChangedEvents()
			End Try
		End Set
	End Property

	Private Sub ClearResources()
		Dim count As Integer = Resources.Count
		Do While count > 0
			Resources.Remove(Resources(0))
			count -= 1
		Loop
	End Sub
	Private Function GenerateResourceIdsString(ByVal xml As String) As ResourceIdCollection
		Dim result As New ResourceIdCollection()
		If String.IsNullOrEmpty(xml) Then
			Return result
		End If

		Return AppointmentResourceIdCollectionXmlPersistenceHelper.ObjectFromXml(result, xml)
	End Function

	Private Function GenerateResourceIdsString() As String
		Dim resourceIds As New ResourceIdCollection()
		Dim count As Integer = Resources.Count
		For i As Integer = 0 To count - 1
			resourceIds.Add(Resources(i).Oid)
		Next i

		Dim helper As New AppointmentResourceIdCollectionXmlPersistenceHelper(resourceIds)
		Return helper.ToXml()
	End Function
End Class